posts/inside-rust/2020-08-12-changes-to-x-py-defaults.md

```diff
@@ -0,0 +1,34 @@
+---
+layout: post
+title: "Changes to x.py defaults"
+author: Joshua Nelson
+team: the compiler team <https://www.rust-lang.org/governance/teams/compiler>
+---
+
+Recently, the defaults for [x.py] - the tool used to [bootstrap] the rust compiler from source - changed. If you regularly contribute to the rust compiler, this might affect your workflow.
+
+## What changes were made?
+
+- The default stage is now dependent on the subcommand.
+  + `dist`: stage 2
+  + `build`: stage 1
+  + `test`: stage 1
+  + `doc`: stage 0
+
+- stage1 `rustc` artifacts are no longer built by `x.py build --stage 1`. To get the old behavior back, use `x.py build --stage 1 src/rustc`.
```

Comment:
Maybe say that `x.py build --stage 1` is equivalent to `x.py build --stage 1 library/libstd`, assuming that's true?

Reply:
Well, sort of. It's 'everything minus rustc', which includes the standard library, tools, and `rustdoc`. I can add that to the post.

Reply:
And most tools are disabled by default, so in the common case it's just `library/std` + `rustdoc`.

Reply:
Hm, that seems like an odd default -- I don't think I ever wanted the tools or rustdoc. But whatever.^^

Reply:
Yeah ... I meant to change it to filter `library/std` in the MCP but by the time I thought of it it had already been accepted :/

Reply:
How hard would it be to change this? If we all agree that to be the sensible choice, maybe we should just do it, even if it requires another MCP.

Reply:
The implementation shouldn't be too hard. I'm happy to open another MCP but since they take at least 10 days to be accepted I think it would be helpful to publish this blog post first.

Reply:
I never opened the MCP 🤦 I'll do that now.

Reply:
https://github.com/rust-lang/compiler-team/issues/351, feel free to second :)